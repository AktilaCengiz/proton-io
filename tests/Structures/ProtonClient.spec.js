/* eslint-disable no-undef */
const ProtonClient = require("../../src/Structures/ProtonClient");
const equal = require("../equal");

const id = "123456789012345678";

test("isOwner string", () => {
    const client = new ProtonClient({
        owners: id,
        intents: []
    });

    equal(client.isOwner(id), false);
    // @ts-ignore
    client.users.cache.set(id, {});
    equal(client.isOwner(id), false);
});

test("isOwner array", () => {
    const client = new ProtonClient({
        owners: [id],
        intents: []
    });

    expect(client.isOwner(id)).toBe(false);
    // @ts-ignore
    client.users.cache.set(id, {});
    equal(client.isOwner(id), false);
    expect(client).toMatchSnapshot();
});
