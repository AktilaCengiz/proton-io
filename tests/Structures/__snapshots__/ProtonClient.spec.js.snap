// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isOwner array 1`] = `
Object {
  "application": null,
  "channels": Array [],
  "guilds": Array [],
  "owners": Array [
    "123456789012345678",
  ],
  "presence": undefined,
  "shard": null,
  "user": null,
  "users": Array [
    "123456789012345678",
  ],
}
`;
